core/src/main/cpp/msg/MessageHeader.cpp

```diff
@@ -433,8 +438,20 @@ MessageHeader::MessageHeader(shared_ptr<MslContext> ctx,
             ? UserAuthenticationData::create(ctx, tokenVerificationMasterToken, headerdata->getMslObject(KEY_USER_AUTHENTICATION_DATA, encoder))
             : shared_ptr<UserAuthenticationData>();
 
-        // Verify the user authentication data.
+        // Identify the user if any.
         if (userAuthData) {
+            // Reject unencrypted messages containing user authentication data.
+            bool encrypted;
```

Comment:
bool encrypted = masterToken ? true : entityAuthData->getScheme().encrypts();
reads a little easier to me, your call though